### ISSUE TITLE
[Suggestion]: Change mod to not be required in a server environment

### ISSUE THREAD

**Issue by totrider** [Mrbysco/RoughlyEnoughProfessions#11]
### General Info

- [ ] I would like to propose new content
- [X] I would like to propose changes
- [ ] I would like to propose another thing

### Suggestion Description

Am currently going through a test of my Modrinth modpack, but REP is causing the server to crash because it is being downloaded for a server environment where REI itself is automatically being omitted.

In the Modrinth.index.json - which is created when you export a modpack for Modrinth - the env field says that server is required, instead of unsupported:

      "path": "mods/RoughlyEnoughProfessions-forge-1.20.1-2.0.2.jar",
      "hashes": {
        "sha1": "5e685b5a74d2ce0a5255822f26238ccca423943c",
        "sha512": "447ae8d96a75898ef4eef902bbe7d652d8ed0adcf323ea85a6965b2f1baec38ab0594d3c9858958173864d72bbed40c695cec974d16bcf25ff1c7663e66f64ec"
      },
      "env": {
        "server": "required",
        "client": "required"
      },
      "downloads": [
        "https://cdn.modrinth.com/data/V8XJ8f5f/versions/qdLND0lY/RoughlyEnoughProfessions-forge-1.20.1-2.0.2.jar"
      ],
      "fileSize": 23204
    },
    
    For reference, this is what REI does:
    
          "path": "mods/RoughlyEnoughItems-12.1.785-forge.jar",
      "hashes": {
        "sha1": "9289cb01c920c1e539034f53ba3736095a2d4410",
        "sha512": "c229232ce96368e16631ac0557caa2b14f3ee2c5d587241da7889b142aecb61eb68dd400f60f368951ffc741f2d375732daf2bf5a46ced239a6e4bb692066110"
      },
      "env": {
        "server": "unsupported",
        "client": "required"
      },
      "downloads": [
        "https://cdn.modrinth.com/data/nfn13YXA/versions/IoS2AjPk/RoughlyEnoughItems-12.1.785-forge.jar"
      ],
      "fileSize": 2384003
    },
    
Can you make it so that REP is also flagged as unsupported automatically through the Modrinth platform? I don't know how or where you do this, but it would make it more convenient to include in modpacks.

**Comment by Mrbysco** [comment]
I've marked server as optional. As it's not unsupported

**Comment by totrider** [comment]
> I've marked server as optional. As it's not unsupported

I don't know how it works, but from what I can see in Modrinth's case, there is only "required" and "unsupported" used across all mod entries in the modrinth.index.json for a modpack I maintain that has tons of mods.

All other REI related mods I see in there have their env for server set to "unsupported".

For context,
Since REI itself is listed as unsupported in a server environment, it wont automatically be installed if say you install a server with a modpack where it is in. This is quite important for simple installs for hosting services, but is also used if you e.g. install a Modrinth modpack with say ATLauncher. If your mod is present, it will automatically crash the server, which is not good.

In other words though, no matter what, if we are talking server environment, the server manager would have to manually install REI itself manually anyway to make this mod work in the first place.

On a side-note, is this something that wont take effect until the mod gets updated? Cause I just worked on an update on my modpack, and this mod still comes up as required on both server and client.

My current working workaround is to manually edit the modrinth.index.json to change the entry to "unsupported", but that is obviously not a great solution.
